activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/impl/TaskRuntimeHelper.java

```diff
@@ -184,5 +192,59 @@ private String getAuthenticatedUser() {
         }
         return internalTask;
     }
+    
+    public Map<String, org.activiti.engine.impl.persistence.entity.VariableInstance> getInternalTaskVariables(String taskId) {
+    	return taskService.getVariableInstancesLocal(taskId);
+    }
+    
+    public void createVariable(boolean isAdmin, CreateTaskVariablePayload createTaskVariablePayload) {	
+    	if (!isAdmin) {
+    		 //Check assignee
+    		getTaskToUpdate(createTaskVariablePayload.getTaskId()); 
+        }
+    	 
+    	if (createTaskVariablePayload.getName() == null || createTaskVariablePayload.getName().isEmpty()) {
+    		throw new IllegalStateException("You cannot create a variable without name");
+    	}
+    	
+    	Map<String, org.activiti.engine.impl.persistence.entity.VariableInstance> variables = taskService.getVariableInstancesLocal(createTaskVariablePayload.getTaskId());
+    	    		
+		//Check if variable already exists
+		if (variables != null && variables.containsKey(createTaskVariablePayload.getName())) {
+			throw new IllegalStateException("Variable already exists");
+		}	
+    	
+		taskService.setVariableLocal(createTaskVariablePayload.getTaskId(),
+									 createTaskVariablePayload.getName(),
+									 createTaskVariablePayload.getValue());	
+    	    
+    }
+    
+    public void updateVariable(boolean isAdmin, UpdateTaskVariablePayload updateTaskVariablePayload) {	
+   	 	if (!isAdmin) {
+   	 		//Check assignee
```

Comment:
@CTI777 @erdemedeiros I think that all user methods should check the assignee. If you are not the assignee of the task you shouldn't be able to see, create or modify variables. So yes.. we should check for assignee here.